### PR TITLE
(PDK-1416) Support Gemfile source option

### PR DESCRIPTION
**Pull request by seanmil**
Add the ability to specify a per-gem source where needed.

### CODE REVIEW

**Comment by scotje** [comment]
@seanmil Thanks for the contribution! Would you mind adding something about this option to the README? Once we have that I think this will be ready to merge.

**Comment by seanmil** [comment]
I don't mind adding something about it, but none of the other gem options are documented anywhere in the PDK that I could see. I would have expected them around here: https://github.com/puppetlabs/pdk-templates#gemfile

How would you think that should look? Another table in the Gemfile section or something else?

**Comment by scotje** [comment]
Indeed the existing options don't appear to be documented at all :)

I agree that the section you linked would be the best place for it. I think if we add an example below the existing table and then a new table that lists the options for each entry that would be good. If you want I can work on adding docs for the existing items but if you want to take a pass at it please feel free!

**Comment by seanmil** [comment]
I've added documentation for all of the options in a gem item, an example in the examples area, and clarified a couple of other points around specifying custom gems which I felt could have been a little clearer. Let me know. Thanks!

File: README.md [hunk 1]
```diff
@@ -156,9 +156,24 @@ Travis uses a .travis.yml file in the root of your repository to learn about you
...
-|required|Allows you to specify gems that are required within the Gemfile. Gems can be defined here within groups, for example we use the :development gem group to add in several gems that are relevant to the development of any module.|
+|required|Allows you to specify gems that are required within the Gemfile. Gems can be defined here within groups, for example we use the :development gem group to add in several gems that are relevant to the development of any module and the :system_tests gem group for gems relevant only to acceptance testing.|
 |optional|Allows you to specify additional gems that are required within the Gemfile. This key can be used to further configure the Gemfile through assignment of a value in the .sync.yml file.|
 
+>Within each Gem group defined using the options above one or more gem item definitions may be listed in an array. Each item in that array must be a gem item hash.
+
+| Gem Item Hash Keys | Description  |
+| :----------------- |:-------------|
+|gem|Required option specifying the gem name.|
+|version|Required option to specify version or range of versions required usin RubyGem version syntax.|
```
Comment (scotje) [inline README.md]:
```suggestion
|version|Required option to specify version or range of versions required using [RubyGem version syntax](https://guides.rubygems.org/patterns/#pessimistic-version-constraint).|
```

Comment (seanmil) [reply README.md]: Good catch on the typo and good idea on the link.

**Comment by scotje** [comment]
Thanks for your contribution to PDK @seanmil!

**Pull request merged**